test/src/test/java/jenkins/model/JenkinsBuildsAndWorkspacesDirectoriesTest.java

```diff
@@ -0,0 +1,364 @@
+package jenkins.model;
+
+import hudson.Functions;
+import hudson.Util;
+import hudson.init.InitMilestone;
+import hudson.maven.MavenModuleSet;
+import hudson.maven.MavenModuleSetBuild;
+import hudson.model.FreeStyleBuild;
+import hudson.model.FreeStyleProject;
+import org.apache.commons.io.FileUtils;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.ExtractResourceSCM;
+import org.jvnet.hudson.test.Issue;
+import org.jvnet.hudson.test.LoggerRule;
+import org.jvnet.hudson.test.MockFolder;
+import org.jvnet.hudson.test.RestartableJenkinsRule;
+import org.jvnet.hudson.test.recipes.LocalData;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.stream.Stream;
+
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+import static org.junit.Assume.assumeFalse;
+
+/**
+ * Since JENKINS-50164, Jenkins#workspacesDir and Jenkins#buildsDir had their associated UI deleted.
+ * So instead of configuring through the UI, we now have to use sysprops for this.
+ * <p>
+ * So this test class uses a {@link RestartableJenkinsRule} to check the behaviour of this sysprop being
+ * present or not between two restarts.
+ *
+ * @since TODO
+ */
+public class JenkinsBuildsAndWorkspacesDirectoriesTest {
+
+    private static final String LOG_WHEN_CHANGING_BUILDS_DIR = "Changing builds directories from ";
+
+    @Rule
+    public RestartableJenkinsRule story = new RestartableJenkinsRule();
+
+    @Rule
+    public LoggerRule loggerRule = new LoggerRule();
+
+    @Before
+    public void before() {
+        clearSystemProperties();
+    }
+
+    @After
+    public void after() {
+        clearSystemProperties();
+    }
+
+    private void clearSystemProperties() {
+        Stream.of(Jenkins.BUILDS_DIR_PROP, Jenkins.WORKSPACES_DIR_PROP)
+                .forEach(System::clearProperty);
+    }
+
+    @Issue("JENKINS-50164")
+    @Test
+    public void badValueForBuildsDir() {
+        story.then((rule) -> {
+            final List<String> badValues = Arrays.asList(
+                    "blah",
```

Comment:
so there are no bad values for windows??  this seems to be unix specific without an assume and may fail but likely not for the expected reason.